### PR TITLE
[QPE-549] tooltip

### PR DESCRIPTION
**Pull request by ahmed-Bazzara**
JIRA ticket
https://tretton37.atlassian.net/secure/RapidBoard.jspa?rapidView=33&projectKey=QPE&modal=detail&selectedIssue=QPE-549

### CODE REVIEW

File: src/tooltip/index.jsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+const getPos = () => {
+  console.log(window.pageXOffset);
```
Comment (kristofferlind) [inline src/tooltip/index.jsx]: remove log

File: src/data-table/data-cell.jsx [hunk 2]
```diff
@@ -61,14 +61,27 @@ function getSemaphoreColors (measurement, semaphoreColors) {
...
   return semaphoreColors.statusColors.normal;
 }
-
-class DataCell extends React.PureComponent {
+class DataCell extends React.Component {
   constructor (props) {
     super(props);
-
+    this.state = {
+      bool: false,
```
Comment (kristofferlind) [inline src/data-table/data-cell.jsx]: rename

File: src/tooltip/index.jsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    }
+    return true;
+  }
+
+  handleRenderTooltip (event) {
+    const { showTooltip } = this.state;
+    const xPosition = event.clientX;
+    const yPosition = event.clientY;
+    this.setState({ showTooltip: !showTooltip,
+      xPosition,
```
Comment (kristofferlind) [inline src/tooltip/index.jsx]: I managed to cause issues where coordinates didn't have time to update, resulting in tooltip showing up over the wrong cell. I think the tooltip position should be fixed, centered above the value.

File: src/tooltip/index.jsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  handleRenderTooltip () {
+    const { showTooltip } = this.state;
+    this.setState({ showTooltip: !showTooltip });
+  }
+
+  handleCalculateTooltipPosition (event) {
+    const tooltipClassName = 'tooltip-wrapper';
+    let tooltip = document.getElementsByClassName(tooltipClassName);
+    const xPositionOffset = 25;
+    const yPositionOffset = 65;
```
Comment (kristofferlind) [inline src/tooltip/index.jsx]: try to replace this with transforms so it works with the dynamic sizes of tooltips (put the transforms in the less file), should be `transform: translate(50%,-100%)` i think

File: src/tooltip/index.jsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+    tooltip[0].style.left = event.clientX - xPositionOffset + 'px';
+    tooltip[0].style.top = event.clientY - yPositionOffset + 'px';
+  }
+
+  render () {
+    const { children, tooltipText, isTooltipActive } = this.props;
```
Comment (kristofferlind) [inline src/tooltip/index.jsx]: is isTooltipActive needed when pointer-events have been disabled?

Comment (ahmed-Bazzara) [reply src/tooltip/index.jsx]: nope it's not needed

**Comment by kristofferlind** [comment]
rebased it so we can move it on to verification

**Pull request merged**